### PR TITLE
Adding CMakeLists.txt and README for the OpenACC Validation and Verif…

### PR DESCRIPTION
**Pull request by Vaidh10**
Adding CMakeLists.txt and README for the OpenACC Validation and Verification Suite so it can be built as part of the LLVM test-suite

### CODE REVIEW

File: External/openacc_vv/README [hunk 1]
```diff
@@ -0,0 +1,22 @@
+OpenACC Validation & Verification Suite
+https://crpl.cis.udel.edu/oaccvv/
+
+This directory contains a CMakeLists.txt for the OpenACC Validation and Verification Suite so it can be built as part
```
Comment (jdenny-ornl) [inline External/openacc_vv/README]: Where possible, please make all text and code fit within 80 columns.  This is required by the LLVM coding standards.

File: External/openacc_vv/README [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+CMAKE_SOURCE_DIR is the root directory of the test-suite sources.
+
+The CMakeLists.txt will search for all C and C++ source files of the
+OpenACC V&V suite, compile and run them. That is, running llvm-lit
+(or "make check") will require a compatible accelerator on the running
+machine.
+
+An example run is:
+
+cmake ../llvm-test-suite -DCMAKE_BUILD_TYPE=Release -DTEST_SUITE_OPENACCVV_ROOT=${HOME}/OpenACCV-V -DTEST_SUITE_LIT=${CLACC_BUILD_DIR}/../build/bin/llvm-lit -DCMAKE_C_COMPILER=${CLACC_BUILD_DIR}/bin/clang -DCMAKE_CXX_COMPILER=${CLACC_BUILD_DIR}/bin/clang++ -DTEST_SUITE_SUBDIRS=External/openacc_vv -DCLACC_BUILD_DIR=${CLACC_BUILD_DIR} -DTEST_SUITE_LIT_FLAGS=-svj1
```
Comment (jdenny-ornl) [inline External/openacc_vv/README]: Please format this line in the same manner as the SOLLVE README does: prefix with `$` and use `\` to fit within 80 columns.

Comment (jdenny-ornl) [inline External/openacc_vv/README]: This command just configures.  Please add the command needed to run the test suite.

File: External/openacc_vv/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+ parallel_reduction.c
+ parallel_scalar_default_firstprivate.c
+ parallel_switch.c
+ parallel_wait.c
+ parallel_wait_devnum.c
+ parallel_while_loop.c
+)
+
+function (add_openaccvv LANG)
+  set(_includedir "${CLACC_BUILD_DIR}/runtimes/runtimes-bins/openmp")
```
Comment (jdenny-ornl) [inline External/openacc_vv/CMakeLists.txt]: Where is `_includedir` used?  If it's not used, then I don't think `CLACC_BUILD_DIR` has to be defined as a cmake variable, as shown in `README`.

File: External/openacc_vv/README (hunk 2 above)
Comment (jdenny-ornl) [inline External/openacc_vv/README]: The name `CLACC_BUILD_DIR` suggests the specified `clang` executable should be the one from the Clacc build directory.  However, the use of `${CLACC_BUILD_DIR}/../build` suggests `CLACC_BUILD_DIR` is actually the Clacc install directory.  I think the latter is the intention.  If so, please rename to `CLACC_INSTALL_DIR`.

File: External/openacc_vv/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+  endif ()
+
+  file(GLOB_RECURSE _tests_sources RELATIVE "${TEST_SUITE_OPENACCVV_ROOT}/Tests" "${TEST_SUITE_OPENACCVV_ROOT}/Tests/*${_langext}" )
+  foreach (_file IN LISTS _tests_sources)
+    get_filename_component(_ext "${_file}" EXT)
+    get_filename_component(_basename "${_file}" NAME_WE)
+    get_filename_component(_directory "${_file}" DIRECTORY)
+    string(REPLACE "." "" _ext "${_ext}")
+    string(REPLACE "/" "_" _directory "${_directory}")
+    string(REPLACE "." "" _directory "${_directory}")
```
Comment (jdenny-ornl) [inline External/openacc_vv/CMakeLists.txt]: Where is `_directory` used?

**Review by jdenny-ornl — CHANGES_REQUESTED** [review]
Thanks for developing this!  Please address my inline review comments, and then I think we can merge this.

File: External/openacc_vv/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  else ()
+    message(FATAL_ERROR "Unsupported languge ${LANG}")
+  endif ()
+
+  file(GLOB_RECURSE _tests_sources RELATIVE "${TEST_SUITE_OPENACCVV_ROOT}/Tests" "${TEST_SUITE_OPENACCVV_ROOT}/Tests/*${_langext}" )
+  foreach (_file IN LISTS _tests_sources)
+    get_filename_component(_ext "${_file}" EXT)
+    get_filename_component(_basename "${_file}" NAME_WE)
+    string(REPLACE "." "" _ext "${_ext}")
+    set(_name "${_basename}-${_ext}")
```
Comment (jdenny-ornl) [inline External/openacc_vv/CMakeLists.txt]: This code transforms test file names like `acc_delete.c` into `acc_delete-c`.  I understand sollve did the same thing, but I don't know why.  Do you?

I find it cumbersome to have to manually perform this transformation whenever I copy and paste a test file name as written in the actual OpenACC V&V suite into, say, `LIT_FILTER`.

Unless it's really needed, I suggest we drop this code and just use `${_file}` where `${_name}` is currently used.

**Review by jdenny-ornl — CHANGES_REQUESTED** [review]
Thanks for applying those changes.  There's one more issue I just noticed.

File: External/openacc_vv/CMakeLists.txt (hunk 5 above)
Comment (Vaidh10) [reply External/openacc_vv/CMakeLists.txt]: I am not entirely sure about the reason for usage of file name transformation in the sollve project. I Updated the code accordingly. Thanks!

**Comment by jdenny-ornl** [comment]
@Vaidh10 Thanks!

**Pull request merged**